app/src/main/java/org/schabi/newpipe/settings/UpdateSettingsFragment.java

```diff
@@ -6,11 +6,22 @@
 
 import org.schabi.newpipe.R;
 
+import static org.schabi.newpipe.CheckForNewAppVersion.startNewVersionCheckService;
+
 public class UpdateSettingsFragment extends BasePreferenceFragment {
     private final Preference.OnPreferenceChangeListener updatePreferenceChange
             = (preference, newValue) -> {
         defaultPreferences.edit()
                 .putBoolean(getString(R.string.update_app_key), (boolean) newValue).apply();
+
+                final boolean shouldWeCheckForUpdate = (boolean) newValue;
+                if (shouldWeCheckForUpdate) {
+                    defaultPreferences.edit()
+                            // -> reset the expire key
+                            .putLong(getString(R.string.update_expiry_key), 0).apply();
+
+                    startNewVersionCheckService();
+                }
```

Comment:
Make it something along the lines of this:
```suggestion
            = (preference, checkForUpdates) -> {
        defaultPreferences.edit()
                .putBoolean(getString(R.string.update_app_key), (boolean) checkForUpdates).apply();

                if ((boolean) checkForUpdates) {
                    // reset the expire time for the update check
                    defaultPreferences.edit().putLong(getString(R.string.update_expiry_key), 0).apply();
                    startNewVersionCheckService();
                }
```